sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -323,6 +354,65 @@ public Instant renewMessageLock(ServiceBusReceivedMessage message) {
         return null;
     }
 
+    /**
+     * Disposes of the consumer by closing the underlying connection to the service.
+     */
+    @Override
+    public void close() {
+        if (isDisposed.getAndSet(true)) {
+            return;
+        }
+
+        final ArrayList<String> keys = new ArrayList<>(openConsumers.keySet());
+        for (String key : keys) {
+            removeLink(key, SignalType.ON_COMPLETE);
+        }
+
+        connectionProcessor.dispose();
+    }
+
+    private Mono<Boolean> isLockTokenValid(UUID lockToken) {
+        final Instant lockedUntilUtc = lockTokenExpirationMap.get(lockToken);
+        if (lockedUntilUtc == null) {
+            logger.warning("lockToken[{}] is not owned by this receiver.", lockToken);
+            return Mono.just(false);
+        }
```

Comment:
Interesting that this returns false instead of throwing an exception and that the lock token being invalid throws an exception.